planner/core/point_get_plan.go

```diff
@@ -330,11 +330,13 @@ func TryFastPlan(ctx sessionctx.Context, node ast.Node) Plan {
 		// Try to convert the `SELECT a, b, c FROM t WHERE (a, b, c) in ((1, 2, 4), (1, 3, 5))` to
 		// `PhysicalUnionAll` which children are `PointGet` if exists an unique key (a, b, c) in table `t`
 		if fp := tryWhereIn2BatchPointGet(ctx, x); fp != nil {
+			if checkFastPlanPrivilege(ctx, fp.dbName, fp.TblInfo.Name.L, mysql.SelectPriv) != nil {
+				return nil
```

Comment:
if err != nil return nil ???

Reply:
We can fall back to the normal plan, then return the error then.

Reply:
Get it!